File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        return solve(cardPoints, k);
    }
private:
    int solve(vector<int>& cardPoints, int k) {
        if(k == cardPoints.size())
            return accumulate(cardPoints.begin(), cardPoints.end(), 0);
        
        int points = 0;
        
        for(int i = 0; i < k; ++i) {
            points += cardPoints[i];
        }
        int end = cardPoints.size() - 1;
        int maxPoints = points;
            
        for(int i = k - 1; i >= 0; --i) {
            points -= cardPoints[i];
            points += cardPoints[end--];
            
            maxPoints = max(maxPoints, points);
        }
        return maxPoints;
    }
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        if k == len(cardPoints):
            return sum(cardPoints)
        
        points = sum(cardPoints[: k])
        
        maxPoints = points
        end = len(cardPoints) - 1
        
        for i in range(k - 1, -1, -1):
            points -= cardPoints[i]
            points += cardPoints[end]
            
            end -= 1
            maxPoints = max(maxPoints, points)
        return maxPoints